src/reducers/importText.ts

```diff
@@ -107,13 +107,20 @@ const importText = (
     const textNormalized = strip(convertedText, { preserveFormatting: true })
 
     // insert the textNormalized into the destValue in the correct place
-    // trim after concatenating in case destValue has whitespace
-    const left = (destValue.slice(0, replaceStart ?? 0) + textNormalized).trimLeft()
     // if cursorCleared is true i.e. clearThought is enabled we don't have to use existing thought to be appended
-    const right = state.cursorCleared ? '' : destValue.slice(replaceEnd ?? 0).trimRight()
-    const newValue = left + right
-
-    const offset = getTextContentFromHTML(left).length
```

Comment:
If `getTextContentFromHTML` is being removed here, please follow our protocol for deleting existing code. Why did that code exist? In what PR was it introduced? What replaces its functionality? What ensures that removing it doesn't introduce any regressions? You can use `git blame` as I showed you to track down the originating commit and PR.

Reply:
Should I include the reasoning in the PR itself?

Reply:
The code existed as a part of commit bf06746e4c26efbed6430a983f17135843976977.

It was introduced in Split thought using browser range api #1203

Previously it was just concatenating left and right variables in importText reducer. However with recent changes we require to paste the text exactly from the caret position . I am still using getTextContentFromHtml now as I realized this has to do with when the pasted text in a html string. So this is necessary. So technicallly I am still using this function.

Reply:
Excellent, thank you!